src/MQTTLibrary/MQTTKeywords.py

```diff
@@ -326,7 +413,9 @@ def _on_message(self, client, userdata, message):
     def _on_message_list(self, client, userdata, message):
         logger.debug('Received message: %s on topic: %s with QoS: %s'
             % (str(message.payload), message.topic, str(message.qos)))
-        self._messages.append(message.payload)
+        if message.topic not in self._messages:
```

Comment:
Since it is now possible to subscribe to multiple topics at once, messages should be cached on a per topic basis instead of in a single list to be able to distinguish topic messages from each other. When the first message on a newly subscribed topic is received, there's no key (nor a list) for that topic in the dictionary, so we create it (the list) if it doesn't already exist for that key so that the following append line always works.

Reply:
👍